Jint/Runtime/Interpreter/JintStatementList.cs

```diff
@@ -68,7 +68,19 @@ public Completion Execute()
                 foreach (var pair in _jintStatements)
                 {
                     s = pair.Statement;
-                    c = pair.Value ?? s.Execute();
+
+                    // @lahma: Am I completely ruining optimization with this check? We need to Execute return statements in debug
+                    // mode, even if they're fast-resolved. Otherwise debug stepping will always step past return statements.
+                    // Could also call OnStep directly here, if value isn't null, and we're in debug mode, but that seems more prone
+                    // to introduce errors with future developments.
+                    if (!_engine._isDebugMode)
```

Comment:
doesn't look too bad, could also skip fast resolve in Initialize when debug mode with comments about the need, I hope tests cover so nobody optimizes the correct behavior away 😉

Reply:
https://github.com/sebastienros/jint/blob/10851bfaca176435bac61cf0d3ed0e6bc3f92476/Jint.Tests/Runtime/Debugger/CallStackTests.cs#L99 does cover it indirectly - will fail if `OnStep` doesn't get called for the `return 'result';` line. But I'll add an explicit test for good measure. Moving debug mode check to `Initialize` - it's also more likely to be spotted there for any future initialization optimizations.